Include restore points in backup/restore section (#700)
timescale/docs.timescale.com-content: using-timescaledb/backup.md

found in the archive. See the PostgreSQL documentation on
[continuous archiving][pg archiving] for more information.

### Restore points [](restore_points)

TimescaleDB natively supports a restore point both for single-node and for
multi-node environments. The restore point can later be used as a recovery target on
single-node or on each node of the entire multi-node cluster, ensuring that it can be
restored to a consistent state.

To create a restore point on a single-node, the [`pg_create_restore_point`][pg_create_restore_point]
function can be used.

On a multi-node cluster, the [`create_distributed_restore_point`][create_distributed_restore_point] function must be used
instead and executed on the access node. In order to achieve synchronization across the multi-node cluster, this function ensures
that the restore point created on the access node is synchronized with each data node. Care should be taken to ensure

> that the data nodes has no locally running user transactions executed without the access node knowledge at the time
> of creating distributed restore point.

Reviewer: ```suggestion
that the data nodes have no locally-running user transactions executed without the access node's involvement when
creating distributed restore point (i.e., by an external user connecting directly to a data node and executing a local
transaction on that data node).
```

found in the archive. See the PostgreSQL documentation on
[continuous archiving][pg archiving] for more information.

### Restore points [](restore_points)

> TimescaleDB natively support a restore point both for single-node and for

Reviewer: ```suggestion
TimescaleDB natively supports a restore point both for single-node and for
```

multi-node environments. The restore point can later be used as a recovery target on
single-node or on each node of the entire multi-node cluster, ensuring that it can be
restored to a consistent state.

> To create a restore point on a single-node [`pg_create_restore_point`][pg_create_restore_point]
> function can be used.

Reviewer: ```suggestion
To create a restore point on a single-node, the [`pg_create_restore_point`][pg_create_restore_point]
function can be used.
```

> To create a named restore point on a single-node [`pg_create_restore_point`][pg_create_restore_point]

Reviewer: What's the advantage of this in single-node?  

I think the above paragraph introduction should be a bit clearer about the user advantage of single vs. multi-node clusters.  In single this (I think) is more about ease-of-use --> just having a specific name to restore to.  In multi-node this is also (mainly?) about correctness/consistency.

Author: The advantage of using restore points in single node is the same as in multi-node, the idea here is that it expected to work in the same way. Maybe we should just concentrate on `distributed restore points` in such case and remove mentioning of the single-node?

> On a multi-node node [`create_distributed_restore_point`][create_distributed_restore_point] function must be used

Reviewer: ```suggestion
On a multi-node cluster, the [`create_distributed_restore_point`][create_distributed_restore_point] function must be used
```

instead and executed on the access-node. In order to achieve synchronization across the multi-node cluster, this function ensures
that the restore point created on the access node is synchronized with each data node. Care should be taken to ensure
third-party solutions properly quiesce the environment before backing up, so that the

> restore point used across nodes does not have outstanding transactions running on the data nodes.

Reviewer: This should be clarified about what is needed here.  I thought the restore point would take a lock.  This makes it sound like something is needed for _correctness_, or are you just worried about lock delay / deadlock?  Can this be clarified?

Author: The basic ideas here is that the function that creates distributed restore point can only guarantee consistency while using TimescaleDB remote transactions. If user have some custom logic and running transactions on a data node directly at the same time (basically without access node knowledge), there are no consistency guarantees for such case.

> Before using restore point make sure PostgreSQL is configured for [WAL Archiving][wal archiving]. To recover to the desired

Reviewer: I think this paragraph is confusing.  It points to a specific section of postgres doc, where instead I think it should also apply to higher-level tools like pgBackrest or WAL-G that use WAL-archiving, no?  I probably would clarify this.

Also, the language could be clearer:

```In order to use restore points, you must configure PostgreSQL to use [WAL archiving][wal archiving], such
that the database stores a copy of all WAL segment, where the restore point will point to a specific
segment (location) in the WAL in which to restore to.  To use a restore point, ...
```
maybe you can finish the paragraph.  It could refer to the same language you currently use, but I think there are probably higher-level pointers to tooks like pgbackrest or WAL-G?

Author: I guess part of this section was related to the data nodes synchronization, mentioned above. I've tried to reformulate the paragraph and reference the third-party tools instead of pointing in to the PG doc
